haskellPackages: update stackage and hackage

cdepillabout opened the pull request:
### This Merge

This PR is the regular merge of the `haskell-updates` branch into `master`.

This branch is being continually built and tested by hydra at https://hydra.nixos.org/jobset/nixpkgs/haskell-updates. You may be able to find an up-to-date Hydra build report at [cdepillabout/nix-haskell-updates-status](https://github.com/cdepillabout/nix-haskell-updates-status).

We roughly aim to merge these `haskell-updates` PRs at least once every two weeks. See the @NixOS/haskell [team calendar](https://cloud.maralorn.de/apps/calendar/p/H6migHmKX7xHoTFa) for who is currently in charge of this branch.

### haskellPackages Workflow Summary

Our workflow is currently described in [`pkgs/development/haskell-modules/HACKING.md`](https://github.com/NixOS/nixpkgs/blob/haskell-updates/pkgs/development/haskell-modules/HACKING.md).

The short version is this:
* We regularly update the Stackage and Hackage pins on `haskell-updates` (normally at the beginning of a merge window).
* The community fixes builds of Haskell packages on that branch.
* We aim at at least one merge of `haskell-updates` into `master` every two weeks.
* We only do the merge if the [`mergeable`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/mergeable) job is succeeding on hydra.
* If a [`maintained`](https://hydra.nixos.org/job/nixpkgs/haskell-updates/maintained) package is still broken at the time of merge, we will only merge if the maintainer has been pinged 7 days in advance. (If you care about a Haskell package, become a maintainer!)

More information about Haskell packages in nixpkgs can be found [in the nixpkgs manual](https://nixos.org/manual/nixpkgs/unstable/#haskell).

---

This is the follow-up to #262645. Come to [#haskell:nixos.org](https://matrix.to/#/#haskell:nixos.org) if you have any questions.

cdepillabout committed:
haskellPackages: stackage LTS 21.16 -> LTS 21.19

This commit has been generated by maintainers/scripts/haskell/update-stackage.sh

cdepillabout committed:
all-cabal-hashes: 2023-10-21T19:49:07Z -> 2023-11-10T11:27:19Z

This commit has been generated by maintainers/scripts/haskell/update-hackage.sh

cdepillabout committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

maralorn committed:
haskell-language-server: Add toplevel jobs for all ghc versions

maralorn commented:
Branch off for 23.11 is in 10 days, November 20th. I hope we can finish this PR until then.

roberth committed:
haskellPackages.releaser: Unmark broken

roberth committed:
haskellPackages: Fixup ormolu minor version in overrides

roberth committed:
haskellPackages: Fixup pandoc minor version in overrides

roberth committed:
haskellPackages: Fixup Cabal minor version in overrides

roberth committed:
haskellPackages: Fixup cabal-install-solver minor version in overrides

roberth committed:
haskellPackages: Fixup fourmolu minor version in overrides

roberth committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

roberth committed:
haskellPackages: Fixup crypton minor version in overrides

roberth committed:
haskellPackages: Fixup hspec-core minor version in overrides

roberth committed:
haskellPackages: Fixup hspec-discover minor version in overrides

roberth committed:
haskellPackages: Fixup hspec-meta minor version in overrides

roberth committed:
haskellPackages: Fixup hspec minor version in overrides

roberth committed:
haskellPackages: Fixup alex minor version in overrides

roberth committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

maralorn committed:
Merge pull request #266708 from hercules-ci/haskell-releaser

haskellPackages.releaser + other fixups

github-actions[bot] committed:
Merge master into haskell-updates

cdepillabout commented:
@libjared Looks like there is a problem in `haskellPackages.sensei`:
- https://hydra.nixos.org/build/240435221
- https://hydra.nixos.org/eval/1801598?filter=haskellPackages.sensei

cdepillabout committed:
haskell.packages.ghc810.text-metrics: jailbreak because of overly-strict bounds

cdepillabout committed:
haskell.packages.ghc88.text-metrics: jailbreak because of overly-strict bounds

cdepillabout committed:
haskell.packages.ghc96.some_1_0_6: update reference from some_1_0_5

cdepillabout committed:
haskellPackages.ostree-pin: disable on darwin

cdepillabout committed:
haskellPackages.rsi-break: disable on darwin

cdepillabout committed:
haskellPackages.vty-windows: disable on all platforms other than windows

maralorn committed:
Merge pull request #266682 from maralorn/hls-toplevel-jobs

haskell-language-server: Add toplevel jobs for all ghc versions

Commit:
haskellPackages.sensei: patch flaky test

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

cdepillabout committed:
Merge branch 'sensei-fix-specs' into haskell-updates

This is from https://github.com/NixOS/nixpkgs/pull/266911

maralorn committed:
Revert "haskell-language-server: Add toplevel jobs for all ghc versions"

This reverts commit 46175bc01df41531e33dbb2123f63b2ddc466650.

Maybe we can find a better solution without polluting the name space that much.

maralorn committed:
haskell-language-server: Fix build

github-actions[bot] committed:
Merge master into haskell-updates

slotThe committed:
haskellPackages.kmonad: unbreak

Version 0.4.2 was released on hackage, which builds correctly.

Closes: https://github.com/NixOS/nixpkgs/issues/266186
Closes: https://github.com/NixOS/nixpkgs/issues/224190

raehik committed:
haskellPackages.ghc96.fgl: update reference 5.8.1.1 -> 5.8.2.0

raehik committed:
haskellPackages.ghc96.th-desugar: update reference 1.15 -> 1.16

github-actions[bot] committed:
Merge master into haskell-updates

cdepillabout committed:
Merge pull request #267477 from raehik/ghc96-pkgs-fgl

haskellPackages.ghc96: update package set pins (fgl, th-desugar)

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

sternenseemann committed:
haskell.compiler: link upstream issue for sphinx 7 patch

sternenseemann committed:
haskellPackages.ghc: 9.4.7 -> 9.4.8

haskell.compiler.ghc94: 9.4.7 -> 9.4.8
haskell.compiler.ghc948: init at 9.4.8

https://www.haskell.org/ghc/blog/20231110-ghc-9.4.8-released.html

sternenseemann committed:
haskell.compiler.integer-simple: construct using include list

There won't be any new integer-simple compilers and they are already in
the minority, so converting the mechanism to an include list is much
better. Maybe native-bignum should now become an exclude list
additionally?

Verified that nix-env -qaP -A haskell.compiler output stays identical.

sternenseemann committed:
Merge remote-tracking branch 'origin/master' into haskell-updates

maralorn committed:
haskell.packages.ghc90.hspec-megaparsec: Fix build

Needed for HLS

github-actions[bot] committed:
Merge master into haskell-updates

github-actions[bot] committed:
Merge master into haskell-updates

roberth committed:
changelog-d: init

Move the overrides to top-level (pkgs/by-name).

roberth committed:
changelog-d: Use justStaticExecutables

I couldn't get the separate bin output to work.

roberth committed:
changelog-d: Move out tests.basic

roberth committed:
changelog-d: Move out updateScript

roberth committed:
haskellPackages.changelog-d: Rename file

roberth committed:
changelog-d: Document tests and updateScript

roberth committed:
changelog-d: Add shell completions

expipiplus1 committed:
haskellPackages: mark builds failing on hydra as broken

This commit has been generated by maintainers/scripts/haskell/mark-broken.sh based on
*evaluation [1801900](https://hydra.nixos.org/eval/1801900) of nixpkgs commit [e98183b](https://github.com/NixOS/nixpkgs/commits/e98183be079f66bc6e880465087741e2284561c0) as of 2023-11-19 15:01 UTC*
from the haskell-updates jobset on hydra under https://hydra.nixos.org/jobset/nixpkgs/haskell-updates

expipiplus1 committed:
Merge remote-tracking branch 'origin/master' into haskell-updates

roberth committed:
Merge pull request #268487 from hercules-ci/changelog-d-static

changelog-d: init / `justStaticExecutables`

github-actions[bot] committed:
Merge master into haskell-updates

expipiplus1 committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

expipiplus1 committed:
haskellPackages.ihaskell: unbreak

expipiplus1 committed:
haskellPackages: add ghc-syntax-highlighter 0.0.10

expipiplus1 committed:
Merge remote-tracking branch 'origin/master' into haskell-updates

expipiplus1 committed:
haskellPackages: mark ihaskell and dependencies unbroken

expipiplus1 committed:
haskellPackages: regenerate package set based on current config

This commit has been generated by maintainers/scripts/haskell/regenerate-hackage-packages.sh

expipiplus1 committed:
haskellPackages: mark builds failing on hydra as broken

This commit has been generated by maintainers/scripts/haskell/mark-broken.sh based on
*evaluation [1801949](https://hydra.nixos.org/eval/1801949) of nixpkgs commit [4b375a4](https://github.com/NixOS/nixpkgs/commits/4b375a4b12e1582709419d5dce44288764145156) as of 2023-11-20 06:46 UTC*
from the haskell-updates jobset on hydra under https://hydra.nixos.org/jobset/nixpkgs/haskell-updates
